app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt

```diff
@@ -16,4 +17,14 @@ sealed class AppAction : Action {
     data class AddNonFatalCrash(val crash: NativeCodeCrash) : AppAction()
     data class RemoveNonFatalCrash(val crash: NativeCodeCrash) : AppAction()
     object RemoveAllNonFatalCrashes : AppAction()
+
+    sealed class MessagingAction : AppAction() {
+        object Initialize : MessagingAction()
+        object Evaluate : MessagingAction()
+        data class UpdateMessageToShow(val message: Message?) : MessagingAction()
```

Comment:
We could fire an `UpdateMessageToShow` action without a message?

Reply:
Yes, I think it will be useful in cases, where we would like to indicate that a previous message needs to be hidden or to indicate no new messages is available

Reply:
You shouldn't need to respond with "no new messages", but _observe_ that the value `messageToShow` is now available and therefore should be displayed.

If the value is consumed (clicked/closed), the `mesageToShow` will be null and the UI will be told to remove anything on the screen.

Reply:
Yeah, for this reason is null-able :).
Do you mean that we need a separated action to consume ?

Reply:
Yeah, so when `MessageClicked` et al actions are observed, we tell the Nimbus storage to mark the message as consumed but we also need to tell the AppStore as well, so that both state solutions are in sync.

In this current design, we would remove the message from the storage and the "messageToShow" would still be available until our next call to `Evaluate` is made. 

> I took out all the logic related to managing the nimbus messages, I will be covering that on https://github.com/mozilla-mobile/fenix/pull/24227

With the middleware/storage gone from this PR, it's a bit harder to review how these parts are connected to others. Maybe these actions and state management should also be moved to the other PR too?

Reply:
Added this commit for easy to review https://github.com/mozilla-mobile/fenix/pull/24225/commits/86dbb8f875c164c5072d71b16929bb7444ee4ae7. I will committed as part of https://github.com/mozilla-mobile/fenix/pull/24227

Reply:
> Yeah, so when MessageClicked et al actions are observed, we tell the Nimbus storage to mark the message as consumed but we also need to tell the AppStore as well, so that both state solutions are in sync.

Sorry for the confusion, these bits are covered in the middleware, we are identifying if the message that was clicked/closed/pressed was the selected one if as we clear the `messageToShow`. 
In this current design, we would remove the message from the storage and the "messageToShow" would still be available until our next call to Evaluate is made.
https://github.com/mozilla-mobile/fenix/pull/24225/commits/86dbb8f875c164c5072d71b16929bb7444ee4ae7#diff-a4f1380abfbf1f76bc6bbd9ef961bd833145c4c75dae3f68dbff419ddd9d5d9cR87-R110